browser/ui/tabs/shared_pinned_tab_service_factory.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_TABS_SHARED_PINNED_TAB_SERVICE_FACTORY_H_
+#define BRAVE_BROWSER_UI_TABS_SHARED_PINNED_TAB_SERVICE_FACTORY_H_
+
+#include "base/memory/singleton.h"
+#include "chrome/browser/profiles/profile_keyed_service_factory.h"
+
+class SharedPinnedTabService;
+
+class SharedPinnedTabServiceFactory : public ProfileKeyedServiceFactory {
+ public:
+  static SharedPinnedTabService* GetForProfile(Profile* profile);
+
+  static SharedPinnedTabServiceFactory* GetInstance();
+
+ private:
+  friend struct base::DefaultSingletonTraits<SharedPinnedTabServiceFactory>;
+
+  SharedPinnedTabServiceFactory();
+  ~SharedPinnedTabServiceFactory() override;
+
+  // BrowserContextKeyedServiceFactory:
+  KeyedService* BuildServiceInstanceFor(
+      content::BrowserContext* profile) const override;
+  bool ServiceIsCreatedWithBrowserContext() const override;
+  content::BrowserContext* GetBrowserContextToUse(
```

Comment:
`GetBrowserContextToUse` is final in `ProfileKeyedServiceFactory` (although in our `chromium_src` file we change it to `final override` and I have no idea what that does :laughing:). Could you use `ProfileBuilder::BuildForAllProfiles` instead? It looks like it selects a separate context for Incognito & Regular profiles.

Reply:
> final override

Yeah, it looks like the file is dropping `final` keyword. Didn't know about the `BuildForAllProfiles`! looks cool. Thanks!

Reply:
Hahahaha goodov told me about it yesterday :laugh